acceptance_modelisation/base_acceptance_map_creator.py

```diff
@@ -436,19 +451,47 @@ def _create_model_cos_zenith_binned(self,
 
         # Determine initial binning value
         cos_zenith_bin = np.sort(np.arange(1.0, 0. - self.initial_cos_zenith_binning, -self.initial_cos_zenith_binning))
-        cos_zenith_observations = [np.cos(obs.get_pointing_altaz(obs.tmid).zen) for obs in observations]
-        run_per_bin = np.histogram(cos_zenith_observations, bins=cos_zenith_bin)[0]
+        cos_zenith_observations = np.array([np.cos(obs.get_pointing_altaz(obs.tmid).zen) for obs in observations])
+        livetime_observations = np.array([obs.observation_live_time_duration.to_value(u.s) for obs in observations])
+        
+        if self.wobble_condition: 
+            wobble_observations = np.array(get_unique_wobble_pointings(observations, self.max_angular_separation))
+        if self.livetime_condition:
+            cut_variable_weights = livetime_observations
+        elif "min_n_observation" in self.cos_zenith_binning_method:
```

Comment:
Ca fonctionne mais est potentiellement instable (voir au dessus).

Reply:
Now there is a check if the method's full string is found in an array with all the possible methods.
The booleans are now based on the array index of the method.